Cython/Includes/libcpp/deque.pxd

```diff
@@ -9,36 +9,109 @@ cdef extern from "<deque>" namespace "std" nogil:
         ctypedef size_t size_type
         ctypedef ptrdiff_t difference_type
 
+        cppclass const_iterator
         cppclass iterator:
-            T& operator*()
+            iterator() except +
+            iterator(iterator&) except +
+            value_type& operator*()
             iterator operator++()
             iterator operator--()
+            iterator operator++(size_type)
+            iterator operator--(size_type)
             iterator operator+(size_type)
             iterator operator-(size_type)
```

Comment:
This adds operators which would be required in C++ to do
```
std::vector<int>::iterator x;
++x;
x++;
x += 5
```
Cython currently appears to ignore whether these operators exist (see https://github.com/cython/cython/issues/4533), so currently they are not strictly necessary in Cython. However it probably still makes sense to define all existing operators since this might change at some point.